src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/TagletWriter.java

```diff
@@ -258,111 +259,119 @@ public static void genTagOutput(TagletManager tagletManager,
                         continue;
                 }
             }
+
             if (taglet instanceof DeprecatedTaglet) {
                 //Deprecated information is documented "inline", not in tag info
                 //section.
                 continue;
             }
+
             if (taglet instanceof SimpleTaglet && !((SimpleTaglet) taglet).enabled) {
                 // taglet has been disabled
                 continue;
             }
-            Content currentOutput = null;
+
             try {
-                currentOutput = taglet.getTagletOutput(element, writer);
-            } catch (UnsupportedTagletOperationException utoe) {
-                //The taglet does not take a member as an argument.  Let's try
-                //a single tag.
-                List<? extends DocTree> tags = utils.getBlockTags(element, taglet);
-                if (!tags.isEmpty()) {
-                    currentOutput = taglet.getTagletOutput(element, tags.get(0), writer);
+                Content tagletOutput = taglet.getAllBlockTagOutput(element, this);
+                if (tagletOutput != null) {
+                    tagletManager.seenTag(taglet.getName());
+                    output.add(tagletOutput);
                 }
-            }
-            if (currentOutput != null) {
-                tagletManager.seenCustomTag(taglet.getName());
-                output.add(currentOutput);
+            } catch (UnsupportedTagletOperationException e) {
+                // malformed taglet:
+                // claims to support block tags but does not provide the appropriate method.
```

Comment:
This deserves a more detailed explanation: what does this claim look like? (It's not obvious from reading this excerpt.)

Reply:
Fixed